lua/barbar/fs.lua

```diff
@@ -21,6 +29,59 @@ function fs.is_relative_path(path)
   return fs.relative(path) == path
 end
 
+--- implementation of certain functions can be simplified based on Neovim version
+if vim.fs then
```

Comment:
In previous discussions we opted to not branch implementation based on Neovim version. However, these functions are baked in to the editor in newer versions, and IMO we should use them if possible.

There are approximate implementations written in the case we can't use them, which does mean there is a potential for slightly altered behavior for lower Neovim versions.